### ISSUE TITLE
test_makeWriteable fails

### ISSUE THREAD

**Issue by sanjayankur31** [pauldmccarthy/fslpy#4]
While updating flspy to 2.2.0 on Fedora 29:

```
platform linux -- Python 3.7.3, pytest-3.6.4, py-1.5.4, pluggy-0.6.0 -- /usr/bin/python3

built on aarch64, little endian

...
______________________________ test_makeWriteable ______________________________
    def test_makeWriteable():
        robuf = bytes(    b'\01\02\03\04')
        wbuf  = bytearray(b'\01\02\03\04')
    
        roarr = np.ndarray((4,), dtype=np.uint8, buffer=robuf)
        warr  = np.ndarray((4,), dtype=np.uint8, buffer=wbuf)
    
        warr.flags['WRITEABLE'] = False
    
        rocopy = dutils.makeWriteable(roarr)
        wcopy  = dutils.makeWriteable(warr)
    
>       assert rocopy.base is not roarr.base
E       AssertionError: assert b'\x01\x02\x03\x04' is not b'\x01\x02\x03\x04'
E        +  where b'\x01\x02\x03\x04' = array([1, 2, 3, 4], dtype=uint8).base
E        +  and   b'\x01\x02\x03\x04' = array([1, 2, 3, 4], dtype=uint8).base
tests/test_fsl_data_utils.py:123: AssertionError
```

The error does not occur on a Fedora 31 (rawhide) system, though:
```
platform linux -- Python 3.7.3, pytest-4.4.1, py-1.8.0, pluggy-0.11.0 -- /usr/bin/python3

built on x86_64
```

The failed build can be seen here on the Fedora build system (the detailed logs for the complete build process are there too): https://koji.fedoraproject.org/koji/taskinfo?taskID=35086803

**Comment by sanjayankur31** [comment]
Update: I ran the build on aarch64, little endian on F31 and it did not fail, so it does not appear to be an architecture related error. For the time being, I've disabled this test on F29.

**Comment by pauldmccarthy** [comment]
Strange  - is it the same version of numpy on all platforms you are testing?

**Comment by sanjayankur31** [comment]
No, the numpy versions are different (newer fedoras have newer versions of software)

- Fedora 31 (where the test works fine) is: 1.16.3
- Fedora 29 (where the test failed) is: 1.15.1

**Comment by pauldmccarthy** [comment]
Right - I added that function to work around changes in numpy 1.16, but evidently did something wrong. I'll try and get a bugfix release out promptly. Thanks!

**Comment by pauldmccarthy** [comment]
Aah, and now I remember what's going on - in versions of numpy prior to 1.16, it is possible to set the `WRITEABLE` flag of a read-only array to True without an error occurring. In fact, older versions of `numpy` will happily mutate a `bytes` object, which is supposed to be immutable!

```python
In [1]: import numpy as np

In [2]: robuf = bytes(b'\01\02\03\04')

In [3]: roarr = np.ndarray((4,), dtype=np.uint8, buffer=robuf)

In [4]: roarr.flags['WRITEABLE'] = True

In [5]: roarr[0] = 255

In [6]: robuf
Out[6]: b'\xff\x02\x03\x04'
```

I'm not sure if I want to build logic into `makeWriteable` to check if the underlying buffer is mutable or not, as I think this would potentially get quite hairy.

I think I'll just defer to the `numpy` behaviour regardless of its version - so if you're using 1.15, it will incorrectly allow you to mutate an immutable buffer.

I'll update the unit test though, so that it passes on older versions of numpy.

**Comment by pauldmccarthy** [comment]
Actually, I've changed my mind - I'll do an `isinstance(array.base, bytes)` test in `makeWriteable` so that at least this case is covered.

**Comment by pauldmccarthy** [comment]
Should be resolved in version 2.3.0!
